### PR TITLE
Fix for timer panicking when duration exceeds max value

### PR DESCRIPTION
**Pull request by wkargul**
It fixes https://github.com/golemfactory/yagna/issues/1951

### CODE REVIEW

File: core/payment/src/api/allocations.rs [hunk 1]
```diff
@@ -202,14 +202,19 @@ pub async fn release_allocation_after(
 ) {
     tokio::task::spawn(async move {
         if let Some(timeout) = allocation_timeout {
-            let timestamp = timeout.timestamp() - Utc::now().timestamp();
-            let mut deadline = 0u64;
+            //FIXME when upgrading to tokio 1.0 or greater. The maximum duration of delay in tokio 0.2 is equal to max_duration(millis).
+            let max_duration = 1 << 36;
```
Comment (nieznanysprawiciel) [inline core/payment/src/api/allocations.rs:206]: It would be safer with explicit type. I'm not sure what type will be inferred here

**Review by nieznanysprawiciel — APPROVED** [review]

**Review by nieznanysprawiciel — COMMENTED** [review]
Please change base branch to `release/v0.10`

File: core/payment/src/api/allocations.rs (hunk 1 above)
Comment (wkargul) [reply core/payment/src/api/allocations.rs]: Ok, it's explicitly typed as i64 now

**Comment by wkargul** [comment]
@nieznanysprawiciel it's tested as requestor and there are no panics anymore

**Review by nieznanysprawiciel — APPROVED** [review]

**Pull request merged**